encryptonize/defs.go

```diff
@@ -34,56 +40,80 @@ type HealthResponse struct {
 type Scope int
 
 const (
-	ScopeRead Scope = iota
-	ScopeCreate
-	ScopeUpdate
-	ScopeDelete
-	ScopeIndex
-	ScopeObjectPermissions
-	ScopeUserManagement
+	ScopeRead              Scope = iota // Scope to decrypt objects
+	ScopeCreate                         // Scope encrypt/store objects
+	ScopeUpdate                         // Scope to update existing objects
+	ScopeDelete                         // Scope to delete stored objects
+	ScopeIndex                          // Scope to list objects
+	ScopeObjectPermissions              // Scope to fetch object permissions
+	ScopeUserManagement                 // Scope to manage users
 )
 
+// CreateUserResponse is the response to a CreateUser call.
 type CreateUserResponse struct {
-	UserID   string `json:"userId"`
+	// ID of the newly created user.
+	UserID string `json:"userId"`
+
+	// Password of the newly created user.
 	Password string `json:"password"`
 }
 
+// CreateGroupResponse is the response to a CreateGroup call.
 type CreateGroupResponse struct {
+	// ID of the newly created group.
 	GroupID string `json:"groupId"`
 }
 
 /////////////////////////////////////////////////////////////////////////
 //                              Encryption                             //
 /////////////////////////////////////////////////////////////////////////
 
+// EncryptResponse is the response to a Encrypt call.
 type EncryptResponse struct {
-	Ciphertext     []byte `json:"ciphertext"`
+	// Encrypted and authenticated data.
+	Ciphertext []byte `json:"ciphertext"`
+
+	// Plaintext authenticated data.
```

Comment:
Probably not for this PR, but is looks confusion that we talk about authenticated data in the comment, but the field is called `AssociatedData`, it would be nice with some explanation for that difference (comment also applicable in other places with associated data)

Reply:
It is a carryover from cryptographic nomenclature, i.e. "authenticated encryption with associated data". Might be a good thing to discuss though.